internal/server/data/grant.go

```diff
@@ -277,8 +273,22 @@ func DeleteGrants(tx WriteTxn, opts DeleteGrantsOptions) error {
 		return fmt.Errorf("DeleteGrants requires an ID to delete")
 	}
 
-	_, err := tx.Exec(query.String(), query.Args...)
-	return err
+	query.B("RETURNING")
+	query.B(columnsForSelect(grantsTable{}))
+
+	rows, err := tx.Query(query.String(), query.Args...)
+	if err != nil {
+		return err
+	}
+
+	grants, err := scanRows(rows, func(g **models.Grant) []any {
+		*g = &models.Grant{}
+		return (*grantsTable)(*g).ScanFields()
+	})
```

Comment:
I forgot to call this out, but this is kind of interesting as well.

We use `RETURNING` with `columnsForSelect` as part of an `UPDATE` statement to get the data about which rows change.

The `scanRows` uses a `**models.Grant` so that the return value is a `[]*models.Grant`, which is what we need to pass to `notifyGrantsChanged`.

In the future if we store destinationId on the grants table we might be able to simply use `RETURNING destination_id`.